pkg/pillar/types/global.go

```diff
@@ -114,6 +124,185 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota + 1
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     uint32
+	intMax     uint32
+	intDefault uint32
+
+	stringDefault   string
+	boolDefault     bool
+	triStateDefault TriState
+}
+
+type ConfigItemSpecMap struct {
+	globalSettings map[string]ConfigItemSpec
+	agentSettings  map[string]ConfigItemSpec
+}
+
+func (configMap *ConfigItemSpecMap) AddIntItem(key string, defaultInt uint32, min uint32, max uint32, agentSetting bool, globalSetting bool) {
+	log.Debugf("Adding int item %s", key)
+	if defaultInt < min || defaultInt > max {
+		log.Fatalf("Adding int item %s failed. Value does not meet given min/max criteria", key)
+	}
+	configItem := ConfigItemSpec{
+		ItemType:   CONFIGITEM_TYPE_INT,
+		Key:        key,
+		intDefault: defaultInt,
+		intMin:     min,
+		intMax:     max,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	}
+	if globalSetting {
+		configMap.globalSettings[key] = configItem
+	}
+	log.Debugf("Added int item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddBoolItem(key string, defaultBool bool, agentSetting bool, globalSetting bool) {
+	log.Debugf("Adding bool item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:    CONFIGITEM_TYPE_BOOL,
+		Key:         key,
+		boolDefault: defaultBool,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	}
+	if globalSetting {
+		configMap.globalSettings[key] = configItem
+	}
+	log.Debugf("Added bool item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddStringItem(key string, defaultString string, agentSetting bool, globalSetting bool) {
+	log.Debugf("Adding string item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:      CONFIGITEM_TYPE_STRING,
+		Key:           key,
+		stringDefault: defaultString,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	}
+	if globalSetting {
+		configMap.globalSettings[key] = configItem
+	}
+	log.Debugf("Added string item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddTriStateItem(key string, defaultTriState TriState, agentSetting bool, globalSetting bool) {
+	log.Debugf("Adding tristate item %s", key)
+	configItem := ConfigItemSpec{
+		Key:             key,
+		ItemType:        CONFIGITEM_TYPE_TRISTATE,
+		triStateDefault: defaultTriState,
+	}
+	if agentSetting {
+		configMap.agentSettings[key] = configItem
+	}
+	if globalSetting {
+		configMap.globalSettings[key] = configItem
+	}
+	log.Debugf("Added tristate item %s", key)
+}
+
+type GlobalConfigItem struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intValue	    uint32
+	stringValue		string
+	boolValue	    bool
+	triStateValue	TriState
+}
+
+type GlobalConfigItemMap struct {
+	globalSettings map[string]GlobalConfigItem
```

Comment:
Didn't we change it to ConfigItem? or ConfigItemValue?

Reply:
Didn't push the change. My bad.